### PR TITLE
chore(lua transform): Add deprecation notice for version 1 API

### PR DESCRIPTION
**Pull request by bruceg**

### CODE REVIEW

File: src/transforms/lua/v1/mod.rs [hunk 1]
```diff
@@ -36,6 +36,9 @@ pub struct LuaConfig {
 
 impl LuaConfig {
     pub fn build(&self) -> crate::Result<Transform> {
+        warn!(
+            "The `lua` transform API version 1 is deprecated. Please convert your script to version 2."
```
Comment (spencergilbert) [inline src/transforms/lua/v1/mod.rs:40]:
```suggestion
            "DEPRECATED The `lua` transform API version 1 is deprecated. Please convert your script to version 2."
```

Per: https://github.com/vectordotdev/vector/blob/master/docs/DEPRECATION.md#deprecation

**Review by spencergilbert — CHANGES_REQUESTED** [review]

**Review by spencergilbert — APPROVED** [review]

File: website/content/en/highlights/2022-10-04-0-25-0-upgrade-guide.md [hunk 2]
```diff
@@ -126,8 +128,14 @@ will be removed in the future.
...
 
 Vector `0.25.0` has introduced distribution of events to multiple endpoints for Elasticsearch sink.
 In order to enable this distribution, a new `endpoints` setting has been introduced which configures one or more destinations to which to distribute the events.
 The existing `endpoint` setting is now deprecated and will be removed in a future version.
+
+#### Deprecation of the Lua version 1 API {#lua-v1-api-deprecation}
+
+Vector `0.9.0` introduced the version 2 API for the `lua` transform.
+This API has long been considered fully mature, obviating the need to maintain the older API.
+The older version 1 API is now deprecated and will be removed in a future version.
```
Comment (jszwedko) [inline website/content/en/highlights/2022-10-04-0-25-0-upgrade-guide.md:141]: Can we add a migration example? AKA a config using version 1 and how to convert that to version 2?

**Review by jszwedko — COMMENTED** [review]
I think we just need an example here, otherwise looks good!

File: website/content/en/highlights/2022-10-04-0-25-0-upgrade-guide.md (hunk 2 above)
Comment (bruceg) [reply website/content/en/highlights/2022-10-04-0-25-0-upgrade-guide.md:141]: Good idea. I'll dig up an old script to convert it.

File: src/transforms/lua/v1/mod.rs [hunk 3]
```diff
@@ -36,6 +36,9 @@ pub struct LuaConfig {
 
 impl LuaConfig {
     pub fn build(&self) -> crate::Result<Transform> {
+        warn!(
+            "DEPRECATED The `lua` transform API version 1 is deprecated. Please convert your script to version 2."
```
Comment (jszwedko) [inline src/transforms/lua/v1/mod.rs:40]: 👍 thanks for following the deprecation policy and adding this warning.

**Review by jszwedko — APPROVED** [review]
Perfect, thanks!

**Pull request merged**